### DISCUSSION TITLE
Widget in ScollView and changing height

### DISCUSSION THREAD

**Discussion by johanfforsberg** [Textualize/textual#210] [Q&A]
Hi,

playing around with textual and I really like it! I've run into an issue though, not sure whether it's a bug or if I'm just doing it wrong.

I'm including a minimal application below. It consists of a widget in a `ScrollView`, which gets updated after one second so that its content takes up more vertical space. The idea is to simulate a case where the data is asynchronously loaded from somewhere else. 

The issue is that only the first row is shown after the update. I guess the `ScrollView` does not update the space needed for the widget. It does work as expected if I skip the `ScrollView`. Also, if I resize the window after the update happened, the widget gets resized to show the entire message, so it seems like some layout refresh is just missing.

```python
import asyncio

from textual.app import App
from textual.widgets import Placeholder, ScrollView
from textual.reactive import Reactive
from textual.widget import Widget

LOREM = """Lorem ipsum dolor sit amet, consectetur adipiscing elit,
sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut
culpa qui officia deserunt mollit anim id est laborum..."""


class LoremWidget(Widget):

    text = Reactive("...loading...")

    async def on_mount(self):
        await asyncio.sleep(1)
        self.text = LOREM

    def render(self):
        return self.text


class LoremApp(App):
    
    async def on_mount(self) -> None:
        self.lorem = ScrollView(LoremWidget())
        await self.view.dock(self.lorem)


if __name__ == "__main__":
    LoremApp.run(log="textual.log")
```

**Comment by geoff-va** [comment]
I think I've run into the same issue as you.

I've got a set of widgets I'd like to cycle through on key press. In the action, I call `self._scrollview.update(self._existing_widget)`. The first widget shown has the proper size. Each other widget after that only shows a few lines in the `ScrollView`, though you can see that the full widget is being cut off.

If I try and resize the terminal, the scrollview properly updates and fills the space. If I don't do any resizing, then the second time I display the widget it's properly sized.

The same thing happens if I use a `WindowView`. The `ScrollView` also looks to use `WindowView` internally so seems it could be related to the `WindowView`?
